From 448ca0d2b599acab2516701e24a98c634f01f8fb Mon Sep 17 00:00:00 2001
From: Wenbin Song <wenbin.song@nxp.com>
Date: Fri, 19 Mar 2021 15:18:09 +0800
Subject: [PATCH 08/18] master/master: fix the issue of sched_setscheduler
 undefined

The symbol "sched_setscheduler" is not exported after linux 5.9.0.

Signed-off-by: Wenbin Song <wenbin.song@nxp.com>
Upstream-Status: Pending
---
 master/master.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/master/master.c b/master/master.c
index f7e5880..b5625d4 100644
--- a/master/master.c
+++ b/master/master.c
@@ -1635,8 +1635,6 @@ static int ec_master_operation_thread(void *priv_data)
  */
 void ec_master_eoe_start(ec_master_t *master /**< EtherCAT master */)
 {
-    struct sched_param param = { .sched_priority = 0 };
-
     if (master->eoe_thread) {
         EC_MASTER_WARN(master, "EoE already running!\n");
         return;
@@ -1661,8 +1659,12 @@ void ec_master_eoe_start(ec_master_t *master /**< EtherCAT master */)
         master->eoe_thread = NULL;
         return;
     }
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+    sched_set_normal(master->eoe_thread, 19);
+#else
+    struct sched_param param = { .sched_priority = 0 };
     sched_setscheduler(master->eoe_thread, SCHED_NORMAL, &param);
+#endif
     set_user_nice(master->eoe_thread, 0);
 }
 
-- 
2.25.1

